packages/db/src/lib/execution.ts

```diff
@@ -191,8 +193,12 @@ export const getWorkflowNameIntersectFromGranuleIds = async (
           .orderBy('timestamp', 'desc');
       }
     });
-  return aggregatedWorkflowCounts
-    .map((workflowCounts: { workflow_name: string }) => workflowCounts.workflow_name);
+
+  return [...new Set(
```

Comment:
Why do we need to do this?   The DB engine/query should be more than capable of selecting distinct results.    (Asking early before I've dived into it)

Reply:
Yeah - I think set wrapping it may be too much of a band-aid versus actually fixing the issue, but am willing to be talked down from that stance if I'm misunderstanding something.

Reply:
I have, for example, isolated the test that produces this and time-gapped each function to make sure there's no race condition being produced by the tests themselves

Reply:
I think the question I was asking is two-fold: 

1)  Why isn't/wasn't this written using `.distinct('workflow_name')`
2)  If this *is* the intention and it can break that way, we should definitely include a unit that demonstrates the intent consistently by writing a test that fails consistently given the current logic. 

Even if we set-fix the unit that's breaking (we should document which test we believe this resolves in the changelog or PR), the test failure shows an inconsistent behavior that we should probably lock down with a test that doesn't behave that way as a function of the test setup.